// apify/utils.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.printOutdatedSdkWarning = exports.logSystemInfo = void 0;
const tslib_1 = require("tslib");
const node_os_1 = require("node:os");
const consts_1 = require("@apify/consts");
const log_1 = tslib_1.__importDefault(require("@apify/log"));
// @ts-expect-error if we enable resolveJsonModule, we end up with `src` folder in `dist`
const package_json_1 = require("@crawlee/core/package.json");
// @ts-expect-error if we enable resolveJsonModule, we end up with `src` folder in `dist`
const package_json_2 = require("apify-client/package.json");
const semver_1 = tslib_1.__importDefault(require("semver"));
// @ts-expect-error if we enable resolveJsonModule, we end up with `src` folder in `dist`
const package_json_3 = require("./package.json");
/**
 * Logs info about system, node version and apify package version.
 * @internal
 */
function logSystemInfo() {
    log_1.default.info('System info', {
        apifyVersion: package_json_3.version,
        apifyClientVersion: package_json_2.version,
        crawleeVersion: package_json_1.version,
        osType: (0, node_os_1.type)(),
        nodeVersion: process.version,
    });
}
exports.logSystemInfo = logSystemInfo;
/**
 * Prints a warning if this version of Apify SDK is outdated.
 * @ignore
 */
function printOutdatedSdkWarning() {
    if (process.env[consts_1.APIFY_ENV_VARS.DISABLE_OUTDATED_WARNING])
        return;
    const latestApifyVersion = process.env[consts_1.APIFY_ENV_VARS.SDK_LATEST_VERSION];
    if (!latestApifyVersion || !semver_1.default.lt(package_json_3.version, latestApifyVersion))
        return;
    log_1.default.warning(`You are using an outdated version (${package_json_3.version}) of Apify SDK. We recommend you to update to the latest version (${latestApifyVersion}).
         Read more about Apify SDK versioning at: https://help.apify.com/en/articles/3184510-updates-and-versioning-of-apify-sdk`);
}
exports.printOutdatedSdkWarning = printOutdatedSdkWarning;
